### DISCUSSION TITLE
Скачивание треков

### DISCUSSION THREAD

**Discussion by hikimorikai** [MarshalX/yandex-music-api#586] [Q&A]
Я еле как разобрался в этой библиотеки, и теперь не знаю как скачать трек, можете мне написать предварительный код по следующему алгоритму
Поиск трека
Выбор трека
Скачивания трека.
Буду очень благодарен!

**Comment by Rirusha** [comment]
Инициализация клиента:
`сlient = Client("ну токен").init()`

Поиск (атрибуты смотри в либе):
`sr = client.search("smth_name")`

Первый трек в выдаче:
`track = sr.tracks.results[0]`

И скачивание:
`track.download("path_to_file.mp3")`

Только не понял, что за выбор...

Да и вообще, брат, читай либу.

**Reply by hikimorikai** [reply]
yandex_music.exceptions.InvalidBitrateError: Unavailable bitrate
Как выбрать битрейт ?...

**Reply by hikimorikai** [reply]
В общем, воспользовался примером Player.py
Но там что-то не так, или я не понял
usage: test.py [-h] [--playlist-name PLAYLIST_NAME] [--skip N] [--shuffle] [--token TOKEN] [--no-save-token]
               [--cache-folder CACHE_FOLDER] [--audio-player AUDIO_PLAYER] [--audio-player-args AUDIO_PLAYER_ARGS]
               [--print-args]
               {likes,user}
test.py: error: the following arguments are required: playlist

**Reply by Rirusha** [reply]
Насчёт первого, могу ссылочку на тг кинуть, там эту проблему как-то разбирали: https://t.me/yandex_music_api/31712
Насчёт второго, в коде player.py глянь, там еще нужно тип плейлиста передать user / like
